packages/edit-site/src/components/navigation-sidebar/navigation-toggle/index.js

```diff
@@ -40,6 +41,15 @@ function NavigationToggle( { icon } ) {
 
 	const disableMotion = useReducedMotion();
 
+	const navigationToggleRef = useRef();
+
+	useEffect( () => {
+		// Focus the trigger button if close
+		if ( ! isNavigationOpen ) {
+			navigationToggleRef.current.focus();
+		}
+	}, [ isNavigationOpen ] );
```

Comment:
Thank you for making this PR! However, I don't think this is the right solution. We don't want to focus the button whenever it gets closed, we only want to focus it whenever route changes. This could have some unwanted side-effects (for instance, if we want to close this one to open another sidebar, the focus could be moved to this button). I opened https://github.com/WordPress/gutenberg/pull/37314 for an alternative approach which I believe could be more robust, but also requires a lot of changes 😅 .